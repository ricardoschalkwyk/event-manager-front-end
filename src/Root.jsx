import React from "react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";

import { store } from "./store";

import App from "./App.jsx";

const Root = () => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <App data-testid="App" />
      </BrowserRouter>
    </Provider>
  );
};

export default Root;
