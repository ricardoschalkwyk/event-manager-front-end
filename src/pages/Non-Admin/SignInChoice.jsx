import { Link, NavLink } from "react-router-dom";
import GoogleLogin from "react-google-login";
import { faEnvelope } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faFacebook } from "@fortawesome/free-brands-svg-icons";

import PropTypes from "prop-types";

import Button from "../../components/forms/Button";
function SignInChoice({ clientId, onSuccess, onFailure }) {
  return (
    <div className="flex h-screen items-center justify-center">
      <div className="text-center">
        <div className="text-gray-200">
          <div className="pb-2 text-3xl font-bold">
            Welcome to Event Management!
          </div>
          <div className="pb-4 text-sm">Sign in with the following:</div>
        </div>
        <div className="mt-2 space-y-4 px-10">
          <GoogleLogin
            className="flex w-full place-content-center space-x-1 rounded bg-red-600 text-white"
            clientId={clientId}
            buttonText="Sign in with Google"
            onSuccess={onSuccess}
            onFailure={onFailure}
            cookiePolicy={"single_host_origin"}
            isSignedIn={true}
          />

          <Button
            bg="bg-blue-500"
            className="flex w-full place-content-center space-x-1 rounded p-2 text-white"
          >
            <div>
              <FontAwesomeIcon icon={faFacebook} />
            </div>

            <div>Facebook</div>
          </Button>

          <Link
            className="flex w-full place-content-center space-x-1 rounded bg-gray-500 p-2 text-white"
            to="/sign-in"
            as={NavLink}
          >
            <div>
              <FontAwesomeIcon icon={faEnvelope} />
            </div>

            <div>Email</div>
          </Link>
        </div>
      </div>
    </div>
  );
}

SignInChoice.propTypes = {
  clientId: PropTypes.string,
  onSuccess: PropTypes.func,
  onFailure: PropTypes.func,
};

export default SignInChoice;
