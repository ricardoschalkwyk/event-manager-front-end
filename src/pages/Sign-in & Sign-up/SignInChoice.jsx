import { useEffect, useState } from "react";
import { Link, NavLink } from "react-router-dom";
import GoogleLogin from "react-google-login";
import { gapi } from "gapi-script";
import { faEnvelope } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faFacebook } from "@fortawesome/free-brands-svg-icons";

import Button from "../../components/forms/Button";
// import Api from "../../api";
function SignInChoice() {
  const [profile, setProfile] = useState([]);
  console.log(profile);

  // const [firstName, setFirstName] = useState(profile.givenName);

  // const [lastName, setLastName] = useState(profile.familyName);

  // const [email, setEmail] = useState(profile.email);

  const clientId =
    "26061662701-q391ntrp908poon72g0gim86pcftfoej.apps.googleusercontent.com";

  const onSuccess = (res) => {
    setProfile(res.profileObj);
  };

  const onFailure = (err) => {
    console.log("failed:", err);
  };

  // const onSignIp = async (firstName, lastName, email) => {
  //   try {
  //     const id_token = googleUser.getAuthResponse().id_token;
  //     // This function fires the post request and the auth endpoint
  //     await Api.post("/auth/google-sign-up", {
  //       firstName,
  //       lastName,
  //       email,
  //     });

  //     navigate("/");
  //   } catch (error) {
  //     setFirstName("");
  //     setLastName("");
  //     setEmail("");
  //   }
  // };

  useEffect(() => {
    const initClient = () => {
      gapi.client.init({
        clientId: clientId,
        scope: "",
      });
    };
    gapi.load("client:auth2", initClient);
  }, []);

  return (
    <div className="flex h-screen items-center justify-center">
      <div className="text-center">
        <div className="text-gray-700">
          <div className="pb-2 text-3xl font-bold">
            Welcome to Event Management!
          </div>
          <div className="pb-4 text-sm">Sign in with the following:</div>
        </div>
        <div className="space-y-4">
          <GoogleLogin
            className="w-full place-content-center"
            clientId={clientId}
            buttonText="Sign in with Google"
            onSuccess={onSuccess}
            onFailure={onFailure}
            cookiePolicy={"single_host_origin"}
            isSignedIn={true}
          />

          <Button
            bg="bg-blue-500"
            padding="p-3"
            className="flex w-full place-content-center space-x-1 rounded-sm text-white shadow-sm shadow-slate-500"
          >
            <div>
              <FontAwesomeIcon icon={faFacebook} />
            </div>

            <div>Facebook</div>
          </Button>

          <Link
            className="flex w-full place-content-center space-x-1 rounded-sm bg-gray-500 p-2.5 text-white shadow-sm shadow-slate-500"
            to="/sign-in"
            as={NavLink}
          >
            <div>
              <FontAwesomeIcon icon={faEnvelope} />
            </div>

            <div>Email</div>
          </Link>
        </div>
      </div>
    </div>
  );
}

SignInChoice.propTypes = {};

export default SignInChoice;
