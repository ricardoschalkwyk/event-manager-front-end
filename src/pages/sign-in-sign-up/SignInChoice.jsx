import { useEffect } from "react";
import { Link, NavLink } from "react-router-dom";
import { faEnvelope } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faFacebook, faGoogle } from "@fortawesome/free-brands-svg-icons";

import { gapi } from "gapi-script";

const googleUrl = () => {
  const options = {
    client_id: import.meta.env.VITE_GOOGLE_CLIENT_ID,
    redirect_uri: `${location.origin}${
      import.meta.env.VITE_GOOGLE_REDIRECT_URI
    }`,
    access_type: "offline",
    response_type: "code",
    prompt: "consent",
    scope: [
      "https://www.googleapis.com/auth/userinfo.email",
      "https://www.googleapis.com/auth/userinfo.profile",
    ].join(" "),
  };

  const query = new URLSearchParams(options);

  return `https://accounts.google.com/o/oauth2/v2/auth?${query}`;
};

// do dis

const facebookUrl = () => {
  const options = {
    client_id: import.meta.env.VITE_FACEBOOK_CLIENT_ID,
    redirect_uri: `${location.origin}${
      import.meta.env.VITE_FACEBOOK_REDIRECT_URI
    }`,

    response_type: "code",
    state: "321cba",
    scope: ["email", "public_profile"].join(" "),
  };

  const query = new URLSearchParams(options);

  return `https://www.facebook.com/v15.0/dialog/oauth?${query}`;
};

function SignInChoice() {
  const client_id = import.meta.env.VITE_GOOGLE_CLIENT_ID;

  console.log(import.meta.env);

  function handleResponse(response) {
    console.log(response.credential);
  }

  useEffect(() => {
    console.log({ url: googleUrl() });

    const initClient = () => {
      gapi.client.init({
        clientId: client_id,
        callback: handleResponse,
        scope: "",
      });
    };
    gapi.load("client:auth2", initClient);
  }, []);

  return (
    <div className="flex h-screen items-center justify-center p-4">
      <div className="text-center">
        <div className="text-gray-700">
          <div className="pb-2 text-3xl font-bold">
            Welcome to Event Management!
          </div>
          <div className="pb-4 text-sm">Sign in with the following:</div>
        </div>
        <div className="space-y-4">
          <div className="flex items-center justify-center gap-2 rounded-md bg-google p-2.5 text-white">
            <FontAwesomeIcon icon={faGoogle} />
            <a href={googleUrl()}>Sign in with Google</a>
          </div>

          <div className="flex items-center justify-center gap-2 rounded-md bg-facebook p-2.5 text-white">
            <FontAwesomeIcon icon={faFacebook} />
            <a href={facebookUrl()}>Sign in with Facebook</a>
          </div>

          <Link
            className="flex w-full place-content-center gap-2 rounded-sm bg-gray-500 p-2.5 text-white shadow-sm shadow-slate-500"
            to="/sign-in"
            as={NavLink}
          >
            <div>
              <FontAwesomeIcon icon={faEnvelope} />
            </div>

            <div>Sign in with Email</div>
          </Link>
        </div>
      </div>
    </div>
  );
}

SignInChoice.propTypes = {};

export default SignInChoice;
