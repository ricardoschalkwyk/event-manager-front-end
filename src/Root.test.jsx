import { render, screen } from "@testing-library/react";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import { describe, expect, it } from "vitest";

import Root from "./Root";
import SignInChoice from "./pages/sign-in-sign-up/SignInChoice";

describe("App tests", () => {
  it("should render Root", () => {
    render(<Root />);
  });

  it("Root renders a App id applied to <App />", () => {
    const { queryByTestId } = render(<Root />);
    expect(queryByTestId("App")).toBeDefined();
  });

  it("should render SignInChoice", () => {
    render(
      <BrowserRouter>
        <Routes>
          <Route path="/sign-in-choice" element={<SignInChoice />} />
        </Routes>
      </BrowserRouter>
    );
  });

  it("Should find id inside SignInChoice", () => {
    expect(screen.queryByTestId("Test-Link")).toBeDefined();
  });

  it("Matches Snapshot", () => {
    const result = render(<Root />);
    expect(result).toMatchSnapshot();
  });
});
