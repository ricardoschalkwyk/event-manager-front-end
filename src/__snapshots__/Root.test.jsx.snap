// Vitest Snapshot v1

exports[`App tests > Matches Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="flex h-screen items-center justify-center p-4"
        >
          <div
            class="text-center"
          >
            <div
              class="text-gray-700"
            >
              <div
                class="pb-2 text-3xl font-bold"
              >
                Welcome to Event Management!
              </div>
              <div
                class="pb-4 text-sm"
              >
                Sign in with the following:
              </div>
            </div>
            <div
              class="space-y-4"
            >
              <a
                class="flex items-center justify-center gap-2 rounded-md bg-google p-2.5 text-white"
                href="https://accounts.google.com/o/oauth2/v2/auth?client_id=26061662701-25vpl3ojbn1t5djbegb9ctiff26vdva7.apps.googleusercontent.com&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Fauth%2Fgoogle&access_type=offline&response_type=code&prompt=consent&scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.profile"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-google "
                  data-icon="google"
                  data-prefix="fab"
                  focusable="false"
                  role="img"
                  viewBox="0 0 488 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
                    fill="currentColor"
                  />
                </svg>
                Sign in with Google
              </a>
              <a
                class="flex items-center justify-center gap-2 rounded-md bg-facebook p-2.5 text-white"
                href="https://www.facebook.com/v15.0/dialog/oauth?client_id=1557189841395800&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Fauth%2Ffacebook&response_type=code&state=321cba&scope=email+public_profile"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-facebook "
                  data-icon="facebook"
                  data-prefix="fab"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
                    fill="currentColor"
                  />
                </svg>
                Sign in with Facebook
              </a>
              <a
                as="[object Object]"
                class="flex w-full place-content-center gap-2 rounded-sm bg-gray-500 p-2.5 text-white shadow-sm shadow-slate-500"
                data-testid="Test-Link"
                href="/sign-in"
              >
                <div>
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-envelope "
                    data-icon="envelope"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48H48zM0 176V384c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V176L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <div>
                  Sign in with Email
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="flex h-screen items-center justify-center p-4"
      >
        <div
          class="text-center"
        >
          <div
            class="text-gray-700"
          >
            <div
              class="pb-2 text-3xl font-bold"
            >
              Welcome to Event Management!
            </div>
            <div
              class="pb-4 text-sm"
            >
              Sign in with the following:
            </div>
          </div>
          <div
            class="space-y-4"
          >
            <a
              class="flex items-center justify-center gap-2 rounded-md bg-google p-2.5 text-white"
              href="https://accounts.google.com/o/oauth2/v2/auth?client_id=26061662701-25vpl3ojbn1t5djbegb9ctiff26vdva7.apps.googleusercontent.com&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Fauth%2Fgoogle&access_type=offline&response_type=code&prompt=consent&scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.profile"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-google "
                data-icon="google"
                data-prefix="fab"
                focusable="false"
                role="img"
                viewBox="0 0 488 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
                  fill="currentColor"
                />
              </svg>
              Sign in with Google
            </a>
            <a
              class="flex items-center justify-center gap-2 rounded-md bg-facebook p-2.5 text-white"
              href="https://www.facebook.com/v15.0/dialog/oauth?client_id=1557189841395800&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Fauth%2Ffacebook&response_type=code&state=321cba&scope=email+public_profile"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-facebook "
                data-icon="facebook"
                data-prefix="fab"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
                  fill="currentColor"
                />
              </svg>
              Sign in with Facebook
            </a>
            <a
              as="[object Object]"
              class="flex w-full place-content-center gap-2 rounded-sm bg-gray-500 p-2.5 text-white shadow-sm shadow-slate-500"
              data-testid="Test-Link"
              href="/sign-in"
            >
              <div>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-envelope "
                  data-icon="envelope"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48H48zM0 176V384c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V176L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
                    fill="currentColor"
                  />
                </svg>
              </div>
              <div>
                Sign in with Email
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
